### ISSUE TITLE
How to use .predict_generator() on new Images - Keras

### ISSUE THREAD

**Issue by debadridtt** [keras-team/keras#11117]
0
down vote
favorite
I've used ImageDataGenerator and flow_from_directory for training and validation. Now I have some new images in a test folder (all images are inside the same folder only), on which I want to predict. But when I use .predict_generator I get:

Found 0 images belonging to 0 class

So I tried these solutions:

1) Keras: How to use predict_generator with ImageDataGenerator? This didn't work out, because its trying on validation set only.

2) How to predict the new image by using model.predict? module image not found

3) How to get predictions with predict_generator on streaming test data in Keras? This also didn't work out.

My train data is basically stored in 4 separate folders, i.e. 4 specific classes, validation also stored in same way and works out pretty well.

So in my test folder I have around 300 images, on which I want to predict and make a dataframe, like this:

      image_name    class
         gghh.jpg       1
         rrtq.png       2
         1113.jpg       1
         44rf.jpg       4
         tyug.png       1
         ssgh.jpg       3
I have also used this following code:

`img = image.load_img(pred_dir, target_size=(28, 28))`
`img_tensor = image.img_to_array(img)`
`img_tensor = np.expand_dims(img_tensor, axis=0)`
`img_tensor /= 255.`

cnn.predict(img_tensor)
But I get this error: [Errno 13] Permission denied: 'D:\\Datasets\\Trell\\images\\new_images\\testing'

But I haven't been able to predict_generator on my test images. So how can I predict on my new images using Keras. I have googled a lot, searched on Kaggle Kernels also but haven't been able to get a solution.

**Comment by grferreira96** [comment]
I was working with a folder "train" and the train had the classes. Have you tried it?
```
test_batches = ImageDataGenerator().flow_from_directory(test_path, target_size=(img_height,img_width), classes=["0", "1"], batch_size=32, class_mode='categorical')

predictions = model.predict_generator(test_batches, steps=1, verbose=0)

test_imgs, test_labels = next(test_batches)
test_labels = test_labels[:,0]
```

**Comment by xzhhzx** [comment]
I guess you can try creating a subfolder under your 'test' folder, i.e. put all your test images under this folder: 'D:\Datasets\Trell\images\new_images\testing\subfolder\'

As for the function flow_from_directory(), you'll always need a subfolder regardless of whether you are training, making validation or testing.

To conclude, your structure of folders might look like this:

\train\
    --\class1\
        --image1
        --image2
        -- ......
    --\class2\
        --image1
        --image2
        -- ......

\validation\
    --\class1\
        --image1
        --image2
        -- ......
    --\class2\
        --image1
        --image2
        -- ......

\test\
    --\subfolder\       ( <- don't care about its name)
        --image1
        --image2
        --image3
        --.......
